src/main/java/frc/robot/Constants.java

```diff
@@ -10,6 +22,216 @@ public static enum Mode {
+    public static final class DIOPorts {
+
+        public static final int switch1 = 1;
+        public static final int switch2 = 2;
     }
+
+    public static final class DriveConstants {
+
+        public static final double MaxSpeedMetPerSec = 6;
+        public static final double MaxAngularRateRadiansPerSec = Math.PI * 2; // 2 PI is one full rotation per second
+        public static final double deadbandPercent = 0.16;
+
+        public static final double trackWidth = Units.inchesToMeters(17.75); // distance between the left and right
+                                                                             // wheels
+        public static final double wheelBase = Units.inchesToMeters(23.75); // distance between the front and rear
+                                                                            // wheels
+        public static final double wheelRadiusM = Units.inchesToMeters(2.02);
+
+        public static final double driveWheelGearReduction = 6.86;
+        public static final double rotationWheelGearReduction = 12.8;
+
+        public static final double frontLeftAngleOffset = 2.922;
+        public static final double frontRightAngleOffset = -1.80;
+        public static final double backLeftAngleOffset = 1.681;
+        public static final double backRightAngleOffset = 1.039;
+
+        public static final double[] driveKps = { 0.7, 0.4, 0.7, 0.7 };
+        public static final double[] driveKds = { 3.5, 2.5, 3.7, 3.5 };
+
+        public static final double[] rotationKps = { 7, 7, 7, 7 };
+        public static final double[] rotationKds = { 0, 0, 0, 0 };
+
+        public static final double driveSnapKp = 1.5;
+        public static final double driveSnapKi = 0;
+        public static final double driveSnapKd = 0;
+
+        public static final SwerveDriveKinematics kinematics = new SwerveDriveKinematics(
+                new Translation2d(trackWidth / 2.0, wheelBase / 2.0), // front left
+                new Translation2d(trackWidth / 2.0, -wheelBase / 2.0), // front right
+                new Translation2d(-trackWidth / 2.0, wheelBase / 2.0), // rear left
+                new Translation2d(-trackWidth / 2.0, -wheelBase / 2.0) // rear right
+        );
+
+        public static final SimpleMotorFeedforward driveFF = new SimpleMotorFeedforward(0.24, 2.185);
+
+        public static final double maxDriveSpeed = 5;
+        public static final double maxTurnRate = 2 * Math.PI;
+
+        public static final double driveJoystickDeadbandPercent = 0.12;
+        public static final double driveMaxJerk = 200.0;
+
+        public static final double poseMoveTranslationkP = 1;
+        public static final double poseMoveTranslationMaxVel = 3;
+        public static final double poseMoveTranslationMaxAccel = 3;
+
+        public static final double poseMoveRotationkP = 0.05;
+        public static final double poseMoveRotationMaxVel = Math.PI;
+        public static final double poseMoveRotationMaxAccel = Math.PI;
+
+    }
```

Comment:
I don't think we need most of these constants if we are using the SwerveAPI. Once we get the drive working, we should remove as much of these as possible to keep it slim

Reply:
I think that robotContainer is initialized, but at the bottom of robotInit?

Reply:
The code is running, I think- SmartDashboard responds when the joysticks are moved

Reply:
Robot.java doesn't show up in this pull request - I would check that this file is correct first